### PR TITLE
Mpuncel/remove use session health

### PR DESCRIPTION
**Pull request by mpuncel**

### CODE REVIEW

**Comment by tummychow** [comment]
on first glance all these changes look good to me so i'm going to :+1:, but I bet we can be even more aggressive. My guess is we could also kill the following:

- kp.RegisterService
- checker.fetchNodeHealth
- checker.consulCheckToResult
- various other code paths in checker that branch on which type of health is being used (eg checker.selectResult and checker.Service can probably be simplified)

anyway great change, it's satisfying to see dead code go

**Comment by anthonybishopric** [comment]
:scissors: :scissors: :ship:

File: pkg/health/checker/checker.go [hunk 1]
```diff
@@ -42,8 +46,7 @@ func NewConsulHealthChecker(opts kp.Options) ConsulHealthChecker {
 	client := kp.NewConsulClient(opts)
 	return consulHealthChecker{
 		client:      client,
-		health:      client.Health(),
-		consulStore: kp.NewConsulStore(opts),
+		consulStore: kp.NewConsulStore(client),
 		WaitTime:    1 * time.Minute,
```
Comment (tummychow) [inline pkg/health/checker/checker.go]: with these changes, is the `client` field still necessary? Also similar to what you did for kp.Store, i'd change this to consume the client directly (if it even still needs it). WaitTime argument should also be safe to remove.

File: pkg/health/checker/checker.go [hunk 2]
```diff
@@ -57,32 +60,17 @@ func (c consulHealthChecker) WatchNodeService(
...
 			return
 		case <-time.After(1 * time.Second):
-			var err error
-			var checks []*api.HealthCheck
-			checks, curIndex, err = c.fetchNodeHealth(nodename, curIndex)
+			kvCheck, err := c.consulStore.GetHealth(serviceID, nodename)
 			if err != nil {
 				errCh <- err
+			} else {
+				resultCh <- consulWatchToResult(kvCheck)
```
Comment (tummychow) [inline pkg/health/checker/checker.go]: blergh polling... luckily only pkg/replication uses this function, and technically it's not _worse_ than it was before, so we can punt on improving this.

**Comment by tummychow** [comment]
:+1: that commit also looks good with the exception of the fields that looked redundant.

File: pkg/kp/fixtures_test.go [hunk 3]
```diff
@@ -10,9 +10,10 @@ import (
 
 type ConsulTestFixture struct {
 	TestServer *testutil.TestServer
-	Store      *consulStore
-	Client     *api.Client
-	T          *testing.T
+	// Store      *consulStore
```
Comment (bwester) [inline pkg/kp/fixtures_test.go]: Delete, don't comment out.

Comment (mpuncel) [inline pkg/kp/fixtures_test.go]: good catch

**Comment by petertseng** [comment]
Think I'm good with my nits now. Just need to fix up those calls to `NewConsulHealthChecker`

**Pull request merged**